### DISCUSSION TITLE
Welcome to finance-quote Discussions!

### DISCUSSION THREAD

**Discussion by bpschuck** [finance-quote/finance-quote#210] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by GitGord** [comment]
Hi,
I've made some changes to the TMX.pm module to, hopefully, make it work but I'm not sure what to do with it now. Are there some tests that are required? I am very new to all this so please be gentle!!

**Reply by bpschuck** [reply]
Thank you for contributing.

How familiar are you with creating forks and pull requests in GitHub?

You should also read the F::Q [Hackers Guide](https://raw.githubusercontent.com/finance-quote/finance-quote/master/Documentation/Hackers-Guide) found in the ./Documentation/ directory of the repository, and give special attention to the section "How to contribute your module to the world".

Thanks again,

- Bruce S

**Reply by GitGord** [reply]
Thanks very much for the reply.
Yes, I found the Hacker Guide, lots of good information. I had a few problems getting Dist::Zilla installed, it kept saying there were modules not installed but they were. I believe it is working now, so I'm getting somewhere.
I have made some changes to TMX.pm and they seem to work, at least as far as the automated tests go and my testing by hand. I would like to write some more tests just to make sure.
I'm not at all familiar with creating forks but I am learning, slowly. I forked F::Q to my repository and I cloned it to my machine here. I created a new project in Pycharm and I'm doing my changes there. It works OK for Perl with a few quirks (I normally use it for Python). Once I'm happy with my changes I'll create the Pull Request, I'm guessing you'll get some sort of notification when I do that?
I'm not sure what the process is from that point. Once you stop laughing at all the bad code I wrote and think it might be OK, does it get merged into the master F::Q branch? How does that propagate to GnuCash? 
Lots more to learn. So far it's been interesting and fun.

**Reply by bpschuck** [reply]
If you notice, installing Dist::Zilla is in a section indicated as _Optional_. For what you are doing, focus on section 5 of the Hackers Guide. Your fork needs to be to a Github repository, and when you create the pull request it automatically creates a notification on the main F::Q repo. If you point me to your fork on GitHub I can give it a look and drop some pointers before you create a pull request.

Once the maintainer team reviews and tests the changes in the PR, we merge it. At some point after a few bug fixes, enhancements, and whatnot, we push a new release to CPAN. GnuCash users need to update their F::Q installation manually  (there are a few ways to do this) for GnuCash to start using it.

Thanks again.
Bruce S.

**Reply by GitGord** [reply]
I guess I missed that about Dist::Zilla... Oh well it was a learning experience! ;)
My Fork is at https://github.com/GitGord/finance-quote
I think it's working OK, I have not done the pull request yet. Please take a look at it and let me know what you think.
Thanks,
Gord

**Reply by bpschuck** [reply]
You are definitely on the right track. But I have a request, not a hard ask, but something we've been trying to do as various modules are updated, fixed, etc.
Instead of executing a `die` for unexpected result or symbol, set the hash value `$info{$symbol, 'success'} = 0`, and supply an error message the `$info{$symbol, 'errormsg'}`. GnuCash does not read, save, or import the stderr coming from invoking the helper app that used F::Q. While it doesn't make use of the info that may be in the `errormsg` key of the returned hash, I believe starting with the GnuCash 5.x release it will have better error reporting for F::Q failures using this data.
Again, not a hard ask, but would be appreciated.
Thank you.
Bruce S.

P.S. **I** probably **should** mention that in the Hacker's Guide...

**Reply by GitGord** [reply]
Just pushed a new version to the repository with what I hope is what you need. Please have a look and let me know.

Gord

**Comment by bpschuck** [comment]
Thanks. I'll take a look today or tomorrow and do some of my own testing.

**Comment by bpschuck** [comment]
@GitGord - Go ahead and create/submit the pull request. There's a couple of minor housekeeping tweaks I'll need to do, but not a problem.
Thank you.

**Reply by GitGord** [reply]
Done. Hope I did it the correct way.
Please let me know if I did something wrong.

Gord
